java/vector/src/test/java/org/apache/arrow/vector/table/BaseTableTest.java

```diff
@@ -180,6 +180,47 @@ void testGetVector() {
     }
   }
 
+  @Test
+  void getVectorCopyByIndex() {
+    List<FieldVector> vectorList = twoIntColumns(allocator);
+    List<FieldVector> vectorList2 = twoIntColumns(allocator);
+    try (Table t = new Table(vectorList)) {
+      // compare value by value
+      for (int vIdx = 0; vIdx < vectorList.size(); vIdx++) {
+        IntVector original = (IntVector) vectorList2.get(vIdx);
+        IntVector copy = (IntVector) t.getVectorCopy(vIdx);
+        assertNotNull(copy);
+        assertEquals(2, copy.getValueCount());
+        assertEquals(0, copy.getNullCount());
+        for (int i = 0; i < t.getRowCount(); i++) {
+          assertEquals(original.getObject(i), copy.getObject(i));
+        }
```

Comment:
not necessarily an issue for this PR, but there is a VectorValueComparator and perhaps it would be good to integrate it as FieldVector#equals if it isn't already

Reply:
Good to know. Thank you